git-github-introduction.md

```diff
@@ -0,0 +1,340 @@
+
+```
+$ git commit -m "Add a README file to the project"
+[new_branch 7660f34] Add a README file to the project
+ 1 file changed, 1 insertion(+)
+  create mode 100644 README.txt
+```
+
+If you want to check to see that all files have been added and committed, type **git status** again. The status should show that the directory is clean, as in the example below:
+
+```
+$ git status
+On branch new_branch
+nothing to commit, working directory clean
+```
+
+To get a list of commit messages made thus far, type **git log**. 
+This will show the user's information and the commit message (in this case "Add a README to the project"):
+
+```
+$ git log
+commit 3c9b1d0068075fdac6ed928f8e5a27ac3253ca5e
+Author: John Smith <john.smith@example.com>
+Date:   Tue Sep 2 14:46:36 2014 -0400
+
+    Add a README to the project.
+```
+
+
+### Push the Commit to the Remote (Forked) Repository
+
+This step will push the branch to the remote repository controlled by the user (the forked version that you created earlier). The changes committed in the "Commit a New File to the Branch" so far exist on the local repository.  The earlier commit message, along with the file, are pushed to the user's remote repository. 
+
+```
+$ git push origin new_branch
+
+Username for 'https://github.com': john-smith 
+Password for 'https://john-smith@github.com': 
+Counting objects: 5, done.
+Delta compression using up to 4 threads.
+Compressing objects: 100% (2/2), done.
+Writing objects: 100% (3/3), 308 bytes | 0 bytes/s, done.
+Total 3 (delta 1), reused 0 (delta 0)
+To https://github.com/john-smith/repy_v2
+ * [new branch]      new_branch -> new_branch
+```
+
+
+### Pull Request
+
+After the files are pushed, they can be made available to the Seattle Testbed Organization, after you (the user) initiates a pull request. 
+
+
+Visit the forked ''repy_v2'' repository and click on the green button next to the drop-down list of branches.
+The next page is where you will preview the commits and write/post a summary or comments about the changes you made. The image below shows where to locate the pull request button.
+
+
+![GitHub Pull Request](/images/pull-request_sm.png)
+
+
+A page (containing the user's Pull Request) is generated by Github and viewable by both the user and developers of the **repy_v2** repository.  A [repy_v2](https://github.com/SeattleTestbed/repy_v2) developer can review the user's changes and either accept, decline, or ask that the user address issues with the pull request.    The pull request's page provides a section to enter comments and allow the user and [repy_v2](https://github.com/SeattleTestbed/repy_v2) developers to collaborate.  The user can make additional changes to the pull request (for example, to address an issue raised) by pushing commits to the user's forked repository.  In the following example, the user first edits the `README.txt` file and saves the changes before issuing **git add README.txt**: 
+

```

Comment:
Also, [the recommended line length is 75 characters](https://github.com/secure-systems-lab/lab-guidelines/commit/1a30af6d1070e7b4e249a8f67d6e05dfec5bed4e#diff-cc90ccfaf9c61945c38d052cf21ed846R62).